### PR TITLE
Vicaruso/new masthead

### PR DESCRIPTION
**Pull request by vincenzocaruso**

### CODE REVIEW

File: lib/components/Masthead/Masthead.tsx [hunk 1]
```diff
@@ -4,167 +4,85 @@ import { MethodNode } from '../../Common';
...
-            key: string;
-            label: string;
-            onClick: Function;
-        };
-        settings: ContentPanelProperties;
-        help: ContentPanelProperties;
-    };
-    userItemAttr?: MastheadAttributes;
+    search?: SearchItem;
+    more: ToolbarItem;
```
Comment (vishwam) [inline lib/components/Masthead/Masthead.tsx]: should more and toolbarItems be optional as well? We might need to remove it for the generic 4xx/5xx pages

File: lib/components/Masthead/Masthead.module.scss [hunk 2]
```diff
@@ -104,6 +147,23 @@ $branding-min-width-size: 120px;
             justify-content: flex-end;
             list-style-type: none;
 
+            .masthead-toolbar-menu ul {
+                list-style: none;
+                padding-left: 0;
+                width: $layout-nav-expanded-width;
+
+                li {
```
Comment (vishwam) [inline lib/components/Masthead/Masthead.module.scss]: add styling through classes? We already define masthead-toolbar-btn

File: lib/components/Masthead/Masthead.tsx [hunk 3]
```diff
@@ -174,33 +92,82 @@ export class Masthead extends React.Component<MastheadProperties, MastheadState>
                         </InlinePopup.Panel>
                     </InlinePopup.Container>
                 }
-                <Attr.div className={cx('masthead-branding', 'inline-text-overflow')} data-test-hook='masthead-application-name'>{this.props.branding}</Attr.div>
-                <Attr.div className={cx('masthead-toolbar-container')}>
+                <Attr.span key={'masthead-branding'} className={cx('masthead-branding', 'inline-text-overflow', { 'forceHideSearch': hidden })} data-test-hook='masthead-application-name'>{this.props.branding}</Attr.span>
```
Comment (vishwam) [inline lib/components/Masthead/Masthead.tsx]: remove data-test-hooks

File: lib/components/Masthead/Masthead.tsx [hunk 4]
```diff
@@ -4,167 +4,85 @@ import { MethodNode } from '../../Common';
...
-                    <ActionTriggerButton
-                        key={key}
-                        attr={{ button: { 'aria-label': item.ariaLabel || key, 'data-test-hook': `masthead-btn-${key}` } }}
-                        icon={item.icon}
-                        onClick={() => this.togglePanel(key)}
-                        className={cx('masthead-toolbar-btn', { 'selected': key === this.state.selectedItem }, item.className)}
+                <li key={`item-${idx}`} className={cx({ 'show-label': !this.props.more.isSelected })}>
+                    < ActionTriggerButton
+                        label={label}
+                        key={label}
```
Comment (vishwam) [inline lib/components/Masthead/Masthead.tsx]: remove? you don't need key here

File: lib/components/Masthead/Masthead.module.scss [hunk 5]
```diff
@@ -90,6 +101,38 @@ $branding-min-width-size: 120px;
...
+    .show-label {
+        display: none;
+
+        @media(min-width: $screen-sm - 1) {
+            display: inherit;
+        }
+    }
+
     .masthead-toolbar-container {
         width: 100% - $masthead-branding-width;
```
[the remark this answers is not in the document]
Comment (vincenzocaruso) [reply lib/components/Masthead/Masthead.module.scss]: both are in % so you can do this

File: lib/components/Masthead/Masthead.module.scss [hunk 6]
```diff
@@ -140,20 +202,11 @@ $branding-min-width-size: 120px;
                     }
                 }
 
-                &.sm {
-                    display: none;
+                span span {
```
[the remark this answers is not in the document]
Comment (vincenzocaruso) [reply lib/components/Masthead/Masthead.module.scss]: you cannot access the label of the action trigger today. this is the only way

File: lib/components/Masthead/Masthead.module.scss (hunk 2 above)
Comment (vincenzocaruso) [reply lib/components/Masthead/Masthead.module.scss]: added a specific class 'masthead-toolbar-btn-container'. This is one level up to masthead-toolbar-btn

File: lib/components/Masthead/Masthead.module.scss [hunk 7]
```diff
@@ -90,6 +101,38 @@ $branding-min-width-size: 120px;
...
     }
 
+    .search-input-container {
+        @media(max-width: $screen-md - 1) {
+            display: none;
+        }
+    }
+
+    .search-input-button {
+        @media(min-width: $screen-md - 1) {
```
[the remark this answers is not in the document]
Comment (vincenzocaruso) [reply lib/components/Masthead/Masthead.module.scss]: this is not redundant. This media query manage the search button to hide or show the search-input. I will add a comment to explain it

File: lib/components/Masthead/Masthead.tsx (hunk 1 above)
Comment (vincenzocaruso) [reply lib/components/Masthead/Masthead.tsx]: good point. added as optional

File: lib/components/SearchInput/SearchInput.module.scss [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+$search-icon-width: $grid-size*8;
+$border-radius-size: $grid-size/2;
+.search-input-container {
+    position: relative;
+    display: inline-flex;
+    margin: 2*$grid-size 5*$grid-size;
+    max-width: $search-max-width;
+    min-width: $search-min-width;
+    width: 100%;
+    height: $search-icon-width;
```
Comment (Coord) [inline lib/components/SearchInput/SearchInput.module.scss]: nit: $search-icon-height

**Pull request merged**